opencog/reasoning/RuleEngine/rules/pln/evaluation-to-member.scm

```diff
@@ -0,0 +1,64 @@
+; ============================================================================= 
+; GeneralEvaluationToMemberRule
+;	EvaluationLink (pred D (ListLink B C))
+;		becomes
+;	MemberLink( B (SatisfyingSetLink( Variable $X 
+;		(EvaluationLink (pred D (ListLink X C))))))
+; -----------------------------------------------------------------------------
+
+(define pln-rule-evaluation-to-member
+	(BindLink
+		(ListLink
+			(VariableNode "$B")
+			(VariableNode "$C")
+			(VariableNode "$D")
+			(TypedVariableLink
+				(VariableNode "$D")
+				(VariableTypeNode "PredicateNode")))
+		(ImplicationLink
+			(EvaluationLink
+				(VariableNode "$D")
```

Comment:
I hope so...

On Wed, Mar 18, 2015 at 11:13 AM, William Ma <notifications@github.com>
wrote:

> In opencog/reasoning/RuleEngine/rules/pln/evaluation-to-member.scm
> <https://github.com/opencog/opencog/pull/1422#discussion_r26635496>:
>
> > +;	MemberLink( B (SatisfyingSetLink( Variable $X
> > +;		(EvaluationLink (pred D (ListLink X C))))))
> > +; -----------------------------------------------------------------------------
> > +
> > +(define pln-rule-evaluation-to-member
> > +	(BindLink
> > +		(ListLink
> > +			(VariableNode "$B")
> > +			(VariableNode "$C")
> > +			(VariableNode "$D")
> > +			(TypedVariableLink
> > +				(VariableNode "$D")
> > +				(VariableTypeNode "PredicateNode")))
> > +		(ImplicationLink
> > +			(EvaluationLink
> > +				(VariableNode "$D")
>
> Wouldn't the (VariableNode "$P") be able to takes care of varying number
> of arguments? When there are more than 1 argument to a Predicate, they are
> always in a ListLink, so (VariableNode "$P") will end up as matching to
> the ListLink, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencog/opencog/pull/1422/files#r26635496>.
>



-- 
Ben Goertzel, PhD
http://goertzel.org

"The reasonable man adapts himself to the world: the unreasonable one
persists in trying to adapt the world to himself. Therefore all progress
depends on the unreasonable man." -- George Bernard Shaw